### PR TITLE
Editoral: Remove unnecessary terminology sections

### PR DESCRIPTION
**Pull request by Honry**
ReSpec knows when something is normative or informative

### CODE REVIEW

**Comment by Honry** [comment]
@marcoscaceres, @kenchris, for the convenience of review, I will split the improvement with new features from ReSpec into several PRs. PTAL. Thanks!

File: index.html [hunk 1]
```diff
@@ -147,49 +147,49 @@
...
-    <dfn data-cite="!HTML#task-source">task source</dfn>,
-    <dfn data-cite="!HTML#the-iframe-element">iframe</dfn>,
-    <dfn data-cite="!HTML#relevant-settings-object">relevant settings object</dfn>,
-    <dfn data-cite="!HTML#active-document">active document</dfn>,
-    <dfn data-cite="!HTML#environment-settings-object">environment settings object</dfn>,
-    <dfn data-cite="!HTML#eventhandler">EventHandler</dfn>,
-    <dfn data-cite="!HTML#focus">focus</dfn>,
-    <dfn data-cite="!HTML#handler-onblur">lose focus</dfn>,
-    <dfn data-cite="!HTML#unloading-document-cleanup-steps">unloading document cleanup steps</dfn>.
+    The terms <dfn data-cite="URL#concept-url">URL</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Please remove all these. ReSpec will auto-link them for you.

File: index.html [hunk 2]
```diff
@@ -147,49 +147,49 @@
...
+    <dfn data-cite="URL#concept-url-host">URL host</dfn>,
+    <dfn data-cite="URL#concept-url-path">URL path</dfn>,
+    <dfn data-cite="URL#concept-url">URL record</dfn>,
+    <dfn data-cite="URL#parse-a-url">parse a URL</dfn>,
+    <dfn data-cite="URL#absolute-url-string">absolute-URL string</dfn>,
+    <dfn data-cite="URL#path-absolute-url-string">path-absolute-URL string</dfn> and
+    <dfn data-cite="URL#concept-basic-url-parser">basic URL parser</dfn> are defined in [[URL]].
+  </p>
+  <p>
+    The following terms are defined in [[HTML]]:
```
Comment (marcoscaceres) [inline index.html]: Same here... remove all HTML redefinitions.

File: index.html [hunk 3]
```diff
@@ -200,98 +200,98 @@
...
+    and <dfn data-cite="PAGE-VISIBILITY#dfn-visibility-states">visibility</dfn>
+    are defined in [[PAGE-VISIBILITY]].
   </p>
   <p>
-    The terms <dfn data-cite="!PERMISSIONS#query-a-permission">query a permission</dfn>,
-    <dfn data-cite="!PERMISSIONS##request-permission-to-use">request permission</dfn>,
-    <dfn data-cite="!PERMISSIONS#dom-permissionstate-granted"><code>"granted"</code></dfn> and
-    <dfn data-cite="!PERMISSIONS##dom-permissionstate-prompt"><code>"prompt"</code></dfn>
-    are defined in [[!PERMISSIONS]].
+    The terms <dfn data-cite="PERMISSIONS#query-a-permission">query a permission</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Remove this one too.

File: index.html [hunk 4]
```diff
@@ -200,98 +200,98 @@
   </p>
   <p>
     The term
-    <dfn data-cite="!MIMESNIFF#mime-type">MIME type</dfn>,
-    <dfn data-cite="!MIMESNIFF#parsing-a-mime-type">Parsing a MIME type</dfn>,
-    <dfn data-cite="!MIMESNIFF#serializing-a-mime-type">Serializing a MIME type</dfn>,
-    <dfn data-cite="!MIMESNIFF#valid-mime-type">valid MIME type string</dfn> and
-    <dfn data-cite="!MIMESNIFF#json-mime-type">JSON MIME type</dfn>
-    are defined in [[!MIMESNIFF]].
+    <dfn data-cite="MIMESNIFF#mime-type">MIME type</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Same here... remove all these.... in the body, do:
```
<body data-cite="MIMESNIFF WEBAPPSEC PERMISSIONS">
```

File: index.html [hunk 5]
```diff
@@ -200,98 +200,98 @@
...
+    <dfn data-cite="MIMESNIFF#parsing-a-mime-type">Parsing a MIME type</dfn>,
+    <dfn data-cite="MIMESNIFF#serializing-a-mime-type">Serializing a MIME type</dfn>,
+    <dfn data-cite="MIMESNIFF#valid-mime-type">valid MIME type string</dfn> and
+    <dfn data-cite="MIMESNIFF#json-mime-type">JSON MIME type</dfn>
+    are defined in [[MIMESNIFF]].
   <p>
-    The term <dfn data-cite="!WEBAPPSEC#secure-context">secure context</dfn>
-    is defined in [[!WEBAPPSEC]].
+    The term <dfn data-cite="WEBAPPSEC#secure-context">secure context</dfn>
```
Comment (marcoscaceres) [inline index.html]: This one can go too...

File: index.html [hunk 6]
```diff
@@ -200,98 +200,98 @@
...
-    <dfn data-cite="!DOM#dictdef-eventinit">EventInit</dfn>,
-    <dfn data-cite="!DOM#eventtarget">EventTarget</dfn>,
-    <dfn data-cite="!DOM#abortsignal">AbortSignal</dfn>,
-    <dfn data-cite="!DOM#abortsignal-aborted-flag">aborted flag</dfn>, and
-    <dfn data-cite="!DOM#abortsignal-add">add the following abort steps</dfn>
-    are defined in [[!DOM]].
+    [[RFC5234]].
+  </p>
+  <p>
+    <dfn data-cite="WEBIDL#dfn-throw"><code>throw</code></dfn>,
```
Comment (marcoscaceres) [inline index.html]: Delete all these too :)

File: index.html [hunk 7]
```diff
@@ -200,98 +200,98 @@
...
+    <dfn data-cite="WEBIDL#syntaxerror"><code>SyntaxError</code></dfn>,
+    <dfn data-cite="WEBIDL#notsupportederror"><code>NotSupportedError</code></dfn>,
+    <dfn data-cite="WEBIDL#networkerror"><code>NetworkError</code></dfn>,
+    <dfn data-cite="WEBIDL#notreadableerror"><code>NotReadableError</code></dfn>,
+    <dfn data-cite="WEBIDL#timeouterror"><code>TimeoutError</code></dfn>, and
+    <dfn data-cite="WEBIDL#notallowederror"><code>NotAllowedError</code></dfn>,
+    are defined in [[WEBIDL]].
+  </p>
+  <p>
+    <dfn data-cite="ECMASCRIPT#sec-promise-objects"><code>Promise</code></dfn>,
```
Comment (marcoscaceres) [inline index.html]: Promise is defined in WebIDL

File: index.html [hunk 8]
```diff
@@ -200,98 +200,98 @@
...
+    <dfn data-cite="ECMASCRIPT#sec-json-object">JSON</dfn>,
+    <dfn data-cite="ECMASCRIPT#sec-json.stringify">JSON.stringify</dfn>,
+    <dfn data-cite="ECMASCRIPT#sec-json.parse">JSON.parse</dfn> and
+    <dfn data-cite="ECMASCRIPT#sec-object-internal-methods-and-internal-slots">internal slots</dfn>
+    are defined in [[ECMASCRIPT]].
+  </p>
+  <p>
+    <dfn data-cite="INFRA#ascii-decode">ASCII decode</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Delete all these

File: index.html [hunk 9]
```diff
@@ -200,98 +200,98 @@
...
+    <dfn data-cite="INFRA#list-is-empty">is empty</dfn>,
+    <dfn data-cite="INFRA#list-is-not-empty">is not empty</dfn>,
+    <dfn data-cite="INFRA#list-append" data-lt="list-append">append</dfn>,
+    <dfn data-cite="INFRA#list-contain" data-lt="list-contain">contains</dfn>,
+    <dfn data-cite="INFRA#parse-json-from-bytes">parse JSON from bytes</dfn> and
+    <dfn data-cite="INFRA#serialize-json-to-bytes">serialize JSON to bytes</dfn>,
+    are defined in [[INFRA]].
+  </p>
+  <p>
+    <dfn data-cite="DOM#concept-event-fire">fire an event</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Same, delete! 🔥🕺

File: index.html [hunk 10]
```diff
@@ -200,98 +200,98 @@
...
+    <dfn data-cite="DOM#interface-event">Event</dfn>,
+    <dfn data-cite="DOM#dictdef-eventinit">EventInit</dfn>,
+    <dfn data-cite="DOM#eventtarget">EventTarget</dfn>,
+    <dfn data-cite="DOM#abortsignal">AbortSignal</dfn>,
+    <dfn data-cite="DOM#abortsignal-aborted-flag">aborted flag</dfn>, and
+    <dfn data-cite="DOM#abortsignal-add">add the following abort steps</dfn>
+    are defined in [[DOM]].
   </p>
   <p>
     <dfn data-cite="ENCODING#concept-stream-push">push</dfn>,
```
Comment (marcoscaceres) [inline index.html]: Delete all these as well... add ENCODING to data-cite on body...

File: index.html [hunk 11]
```diff
@@ -1053,7 +1053,7 @@
     is used for acquiring trust for a given operation.
   </p>
   <p class="note">
-    The Permissions API [[!PERMISSIONS]] is suggested to be used by
+    The Permissions API [[PERMISSIONS]] is suggested to be used by
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
    The [[[PERMISSIONS]]] is suggested to be used by
```

File: index.html [hunk 12]
```diff
@@ -2227,7 +2227,7 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
                 If it resolved with <code><a>"granted"</a></code>
                 (i.e. an <a>expressed permission</a> has been granted
                 to the <a>origin</a> and <a>global object</a> using
-                the Permissions API [[!PERMISSIONS]]), return <code>true</code>.
+                the Permissions API [[PERMISSIONS]]), return <code>true</code>.
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
                the [[[PERMISSIONS]]]), return <code>true</code>.
```

File: index.html [hunk 13]
```diff
@@ -2431,7 +2431,7 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
           <li>
             Set the <var>ndef</var>'s PAYLOAD field to
             <var>record</var>'s data encoded according to the
-            [[!NFC-STANDARDS]], NFC Forum Text Record Type
+            [[NFC-STANDARDS]], NFC Forum Text Record Type
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            [[[NFC-STANDARDS]]], NFC Forum Text Record Type
```

Comment (marcoscaceres) [reply index.html]: Three [[[ "expands" a reference...

File: index.html [hunk 14]
```diff
@@ -3116,7 +3116,7 @@ <h3>Receiving and parsing content</h3>
               </li>
             </ol>
             <p class="note">
-              Using the encoder from [[!ENCODING]], it is only possible
+              Using the encoder from [[ENCODING]], it is only possible
```
Comment (marcoscaceres) [inline index.html]: Can you do?: 

```suggestion
              Using the <a>encoder</a>, it is only possible
```

File: index.html [hunk 15]
```diff
@@ -3116,7 +3116,7 @@ <h3>Receiving and parsing content</h3>
...
               to encode as UTF-8, unless you do the encoding manually
               and use the <code>"opaque"</code>
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
              and use the `"opaque"`
```

**Comment by marcoscaceres** [comment]
@Honry, you can start with the simple "!", but you will need to deal with deleting most of the terminology section... do you want to do that now?

**Comment by marcoscaceres** [comment]
Also, ReSpec will start complaining that it can't find the right terms... you can use this to help you find what you need:
https://respec.org/xref/

**Comment by Honry** [comment]
> @Honry, you can start with the simple "!", but you will need to deal with deleting most of the terminology section... do you want to do that now?

Yeah, I'm working on it.

File: index.html (hunk 8 above)
Comment (Honry) [reply index.html]: @marcoscaceres, do I need to add the "INFRA" to the `<body>`? It's odd, as without it in the `<body>`, all the `INFRA` relevant links disappeared. e.g. `[=list/For each=]` throws ReSpec error.

While this does not happen in WakeLock, did I miss something critical?

Comment (Honry) [reply index.html]: I found the root cause, `xref: "web-platform"` is required in `respecConfig`. :)

Comment (marcoscaceres) [reply index.html]: was just about to reply :) Exactly.

Comment (marcoscaceres) [reply index.html]: so, here is what specs "web-platform" maps to: 
https://github.com/w3c/respec/blob/84fef3fea275143349a313a888546e8f2cb84137/src/core/xref.js#L24

I'm thinking we should add Encoding and MIME there.

Comment (Honry) [reply index.html]: Thanks, good to know that!

Comment (Honry) [reply index.html]: > `typedef (DOMString or unrestricted double or ArrayBuffer or Dictionary) NDEFRecordData;
`

@marcoscaceres, the `Dictionary` type in above IDL could not be recognized, any idea?

Comment (marcoscaceres) [reply index.html]: Dictionary is not a type :) You need an actually type.

Comment (marcoscaceres) [reply index.html]: Put differently, what "dictionary"? Or do you mean "object"... 

See https://heycam.github.io/webidl/#prod-DistinguishableType

Comment (Honry) [reply index.html]: It expects a JSON object, and does not indicate any "dictionary". For this situation, can it be an "Object"? I'm not quite sure, tbh.

Comment (marcoscaceres) [reply index.html]: no, it would have to be "object" (lowercase).

File: index.html (hunk 15 above)
Comment (Honry) [reply index.html]: I'd like to unify this in a separate PR. :)

**Comment by Honry** [comment]
@marcoscaceres, all done! PTAL again, thanks!

File: index.html (hunk 15 above)
Comment (marcoscaceres) [reply index.html]: sure, sounds good.

File: index.html [hunk 16]
```diff
@@ -147,151 +141,18 @@
     contains.
   </p>
   <p>
-    Implementations that use [[!ECMAScript]] to implement the APIs defined in
+    Implementations that use [[ECMAScript]] to implement the APIs defined in
```
Comment (marcoscaceres) [inline index.html:145]: Separate PR... but delete this whole statement. It serves no purpose and I don't know why people keep including it.

Comment (marcoscaceres) [reply index.html:145]: (I know also you didn't add it :) ... just constantly having to delete it from specs!)

Comment (Honry) [reply index.html:145]: For sure!

File: index.html [hunk 17]
```diff
@@ -403,7 +264,7 @@
       </tr>
       <tr>
         <td>1</td>
-        <td>NFC Forum <dfn>well-known type</dfn> as defined in [[!NFC-RTD]]
+        <td>NFC Forum <dfn>well-known type</dfn> as defined in [[NFC-RTD]]
```
Comment (marcoscaceres) [inline index.html]: We add this to be polite... and to remind other editors to never export these:

```suggestion
        <td>NFC Forum <dfn data-no-export="">well-known type</dfn> as defined in [[NFC-RTD]]
```

File: index.html [hunk 18]
```diff
@@ -836,17 +698,17 @@
       <ol>
         <li>
           Reading an <a>NFC tag</a> containing a <a>Web NFC message</a>, when a
-          web page using the Web NFC API is open and in <a>focus</a>.
+          web page using the Web NFC API is open and in <a data-cite="html#focus">focus</a>.
```
Comment (marcoscaceres) [inline index.html]: what does the xref UI say about how to cite this?

Comment (marcoscaceres) [reply index.html]: oh, we integrated the UI into ReSpec today (do a "shift-refresh")... if you click on the ReSpec pill, you can access the UI by clicking "Search Definitions".

Comment (marcoscaceres) [reply index.html]: Oh, it doesn't show up... it means this is not exported! That means something might be wrong... either HTML needs to export it or you have a spec bug! 🚨

File: index.html [hunk 19]
```diff
@@ -416,7 +277,7 @@
       </tr>
       <tr>
         <td>4</td>
-        <td>NFC Forum <dfn>external type</dfn> as defined in [[!NFC-RTD]]</td>
+        <td>NFC Forum <dfn>external type</dfn> as defined in [[NFC-RTD]]</td>
```
Comment (marcoscaceres) [inline index.html]: As above...

File: index.html (hunk 18 above)
Comment (marcoscaceres) [reply index.html]: Yes, spec bug! Two issues. You want this one instead:
https://html.spec.whatwg.org/multipage/interaction.html#focused

But it's not exported... you need to find out why? And if you think it should be exported from HTML, you need to ask the HTML folks to export it (or send them a PR, which adds `data-export=""` to the `dnf` in the HTML spec).

File: index.html [hunk 20]
```diff
@@ -1071,22 +933,24 @@
...
         This also means that UAs should block access to the NFC radio if
         the display is off or the device is locked.
         For backgrounded web pages, receiving and pushing <a>NFC content</a>
         must be <a id="#nfc-suspended">suspended</a>.
       </li>
       <li>
-        When pushing <a>Web NFC content</a>, the <a>serialized origin</a> and
-        the <a>URL path</a> of the <a>current settings object</a> when
+        When pushing <a>Web NFC content</a>, the
+        <a data-cite="html#ascii-serialisation-of-an-origin">serialized origin</a>
```
Comment (marcoscaceres) [inline index.html]: Same here... find out why this was not exported from HTML if you need it. Maybe there is something in the URL spec you can use.

File: index.html [hunk 21]
```diff
@@ -1572,7 +1437,8 @@ <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
     is available. The <dfn>NFCReadingEvent.message</dfn> is an <a>NDEFMessage</a> object.
     <dfn>NFCReadingEventInit</dfn> is used to initialize a new event with a serial number
     and the <a>NDEFMessageInit</a> data via the <dfn>NFCReadingEventInit.message</dfn> member.
-    If <dfn>NFCReadingEventInit.serialNumber</dfn> is <a>not present</a> or is `null`,
+    If <dfn>NFCReadingEventInit.serialNumber</dfn> is
```
Comment (marcoscaceres) [inline index.html:1650]: maybe different PR, but this seems like a very odd place to define something.

File: index.html [hunk 22]
```diff
@@ -1572,7 +1437,8 @@ <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
...
+    <a data-cite="webidl#dfn-present">not present</a> or is `null`,
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
    not <a>present</a> or is `null`,
```

File: index.html [hunk 23]
```diff
@@ -1582,9 +1448,9 @@ <h2>The <dfn>NDEFRecordType</dfn> enum</h2>
   </p>
   <p>
     The <dfn>NFCErrorEvent</dfn> is the event being dispatched on errors,
-    with the <code><a>DOMException</a></code> object as the <dfn>NFCErrorEvent.error</dfn>
+    with the {{DOMException}} object as the <dfn>NFCErrorEvent.error</dfn>
     attribute. <dfn>NFCErrorEventInit</dfn> is used in order to initialize a new
-    event with a <code><a>DOMException</a></code> data via the <dfn>NFCErrorEventInit.error</dfn>
+    event with a {{DOMException}} data via the <dfn>NFCErrorEventInit.error</dfn>
```
Comment (marcoscaceres) [inline index.html:1663]: Different pull request/issue... but attaching errors to events is a bad code smell.

File: index.html [hunk 24]
```diff
@@ -1986,22 +1854,21 @@ <h3>The <dfn>NFCReaderOptions</dfn> dictionary</h3>
           <li>
             If there is no underlying <a>NFC Adapter</a>, or if a connection cannot
             be established, then reject <var>p</var> with a
-            <code>"<a>NotReadableError</a>"</code> <code><a>DOMException</a></code>
+            "{{NotReadableError}}" {{DOMException}}
```
Comment (marcoscaceres) [inline index.html]: Same for the rest... landed support for this today!:

```suggestion
            {{"NotReadableError"}} {{DOMException}}
```

File: index.html (hunk 20 above)
Comment (Honry) [reply index.html]: I can find this in  xref UI, as it uses a different name(serialized origin), that's why we need a "data-cite" here, right?

File: index.html (hunk 21 above)
Comment (Honry) [reply index.html:1650]: Exactly.

File: index.html (hunk 20 above)
Comment (marcoscaceres) [reply index.html]: Nope :) it means there is a spec bug. If something hasn’t been exported from HTML, that warrants further investigation. Basically, we need to treat every data-cite as a bug 🕷 it should only ever be needed for specs that haven’t exported their terms.

Comment (marcoscaceres) [reply index.html]: Oh wait, I totally misread. You are correct!

Comment (marcoscaceres) [reply index.html]: But, don’t use “data-cite”, use data-lt="serialization of an origin"

Comment (Honry) [reply index.html]: Thank you, tips learned.

File: index.html (hunk 18 above)
Comment (Honry) [reply index.html]: OK, I will drop a PR to the HTML spec.

Comment (Honry) [reply index.html]: Done, at https://github.com/whatwg/html/pull/4811

File: index.html (hunk 21 above)
Comment (Honry) [reply index.html:1650]: Files https://github.com/w3c/web-nfc/issues/252.

**Review by marcoscaceres — APPROVED** [review]
Approving in case you want to merge what you have  so far...

**Comment by Honry** [comment]
@marcoscaceres, big thanks! :)  @kenchris, could you help land this?

**Comment by marcoscaceres** [comment]
Keep up the great work @Honry. Looking forward to the next set.

**Comment by Honry** [comment]
@kenchris, rebase done! Please help land it. :)

**Pull request merged**